src/shoppinglist.cpp

```diff
@@ -0,0 +1,173 @@
+#include "shoppinglist.h"
+
+///Number of properties associated with each shopping list item (price, name, amount)
+const int NUM_SHOPPINGLIST_PROPERTIES = 3;
+///Column index for name in shopping list model
+#define ITEM_NAME_COL 0
+///Column index for price in shopping list model
+#define ITEM_PRICE_COL 1
+///Column index for amount in shopping list model
+#define ITEM_AMOUNT_COL 2
```

Comment:
The ordering here defines the order of the items are they would be displayed in a QTableView. It's a pure display ordering, doesn't affect the internal data or make assumptions on the data ordering.

The display ordering is mentioned in the documentation for data() (line 56-63), but this seems to be consistent with these defines.